mixer/pkg/runtime2/runtime.go

```diff
@@ -141,6 +141,9 @@ func (c *Runtime) StopListening() {
 	if c.shutdown != nil {
 		c.shutdown <- struct{}{}
 		c.shutdown = nil
+		if err := c.store.Close(); err != nil {
```

Comment:
Should the close be before c.waitQuiesceListening, or after? 
The background go-routine will be listening to both the c.shutdown channel, as well as the channel returned by the store. When the store is closed here, what will happen to the channel that is returned by the store? (Or

Reply:
Ooh, nice catch!  The watch channel shall be closed on store's `Close()`, that means the ordering is incorrect. `watchChanges` might not finish at all. Changed the order, waitQuiesceListening first, and then store.Close. Thank you!